pkg/services/object/get/assemble.go

```diff
@@ -44,6 +44,14 @@ func (exec *execCtx) assemble() {
 		}
 	}
 
+	if splitInfo.SplitID() == nil {
+		exec.log.Debug("handling V2 split")
+		exec.processV2Split(splitInfo)
```

Comment:
i thought we wont need completely separate branch of code for this (schemes differ by LINK obj format and quick access to piece sizes). But ok, done as done

Reply:
i wanted them to be separate to drop v1 some time in the future

also, i find it more readable. storing some internal state (v1/v2 split) is harder is read IMO. assembly code it complex enough